docs/source/API/core/policies/TeamThreadRange.rst

```diff
@@ -0,0 +1,93 @@
+``TeamThreadRange``
+===================
+
+.. role::cpp(code)
+    :language: cpp
+
+Header File: ``<Kokkos_Core.hpp>``
+
+Usage
+-----
+
+.. code-block:: cpp
+
+    parallel_for(TeamThreadRange(team,range), [=] (int i) {...});
+    parallel_reduce(TeamThreadRange(team,begin,end), 
+        [=] (int i, double& lsum) {...},sum);
+
+TeamThreadRange is a `nested execution policy <./NestedPolicies.html>`_ used inside hierarchical parallelism. In contrast to global policies, the public interface for nested policies is implemented as functions, in order to enable implicit templating on the execution space type via the team handle.
+
+Synopsis 
+--------
+
+.. code-block:: cpp
+    
+    template<class TeamMemberType, class iType>
+    /* implementation defined */ TeamThreadRange(TeamMemberType team, iType count);
+    
+    template<class TeamMemberType, class iType1, class iType2>
+    /* implementation defined */ TeamThreadRange(TeamMemberType team, iType1 begin, iType2 end);
+
+Description
+-----------
+
+.. cpp:function:: template<class TeamMemberType, class iType> TeamThreadRange(TeamMemberType team, iType count);
+
+.. /* Implementation defined */
+
+Splits the index range ``0`` to ``count-1`` over the threads of the team. 
+
+* **Arguments**  
+    - ``team``: a handle to the calling team execution context.
+    - ``count``: index range length. 
+
+* **Returns**    
+    - Implementation defined type.
+
+* **Requirements**   
+    - ``TeamMemberType`` is a type that models `TeamHandle <./TeamHandleConcept.html>`_
+    - ``std::is_integral<iType>::value`` is true.
+    - Every member thread of ``team`` must call the operation in the same branch, i.e. it is not legal to have some threads call this function in one branch, and the other threads of ``team`` call it in another branch.
+    - ``count >= 0`` is true;
```

Comment:
all of the above should be indented so that it is under the corresponding constructor